CondFormats/JetMETObjects/src/SimpleJetCorrector.cc

```diff
@@ -38,7 +38,8 @@ float SimpleJetCorrector::correction(const std::vector<float>& fX,const std::vec
   float result = 1.;
   float tmp    = 0.0;
   float cor    = 0.0;
-  int bin = mParameters.binIndex(fX);
+  int bin = -1;
+  bin = (fX.size()<=3 && fX.size()>0) ? mParameters.binIndexN(fX) : mParameters.binIndex(fX);
```

Comment:
I want to second what @kpedro88 said. An additional reason is that we want to use the new algorithm for all the dimensions it has been tested for (1,2,3), but right now we are using the original, brute-force algorithm as a catch-all for untested numbers of dimensions. I would expect the new algorithm, binIndexN, to have much better performance than this original, so this limit can always be raised/removed if and when we find it necessary.